src/utils.cpp

```diff
@@ -477,21 +478,70 @@ std::string exec_system(const char* cmd)
   return result;
 }
 
-std::vector<std::string> resolve_binary_path(const std::string& cmd)
+std::vector<std::string> tokenize(std::string s, std::string delimiter)
 {
-  std::string query;
-  query += "command -v ";
-  query += cmd;
-  std::string result = exec_system(query.c_str());
+  std::vector<std::string> paths;
+  size_t start = 0, end = 0;
 
-  if (result.size())
+  while (end != std::string::npos)
   {
-    return split_string(result, '\n');
+    end = s.find(delimiter, start);
+    auto path = s.substr(start, (end == std::string::npos) ? std::string::npos : end-start);
+    paths.push_back(path);
+    start = (end > std::string::npos-delimiter.size()) ? std::string::npos : end+delimiter.size();
   }
-  else
+
+  return paths;
+}
+
+std::vector<std::string> expand_wildcard_path(const std::string& path)
+{
+  glob_t glob_result;
+  memset(&glob_result, 0, sizeof(glob_result));
+
+  if (glob(path.c_str(), GLOB_NOCHECK, NULL, &glob_result)) {
+    globfree(&glob_result);
+    throw std::runtime_error("glob() failed");
+  }
+
+  std::vector<std::string> matching_paths;
+  for (size_t i = 0; i < glob_result.gl_pathc; ++i) {
+    matching_paths.push_back(std::string(glob_result.gl_pathv[i]));
+  }
+
+  globfree(&glob_result);
+  return matching_paths;
+}
+
+std::vector<std::string> expand_wildcard_paths(const std::vector<std::string> paths)
+{
+  std::vector<std::string> expanded_paths;
+  for (std::string p : paths)
   {
-    return { cmd };
+    auto ep = expand_wildcard_path(p);
+    expanded_paths.insert(expanded_paths.end(), ep.begin(), ep.end());
   }
+  return expanded_paths;
+}
+
+std::vector<std::string> resolve_binary_path(const std::string& cmd)
```

Comment:
Will do. I think resolve_binary_path itself is covered well by the runtime tests, but I'll add tests for the others.

Reply:
I think it will be good to have some tests for resolve_binary_path as well. Runtime tests are slow to run and it's trick to test all possible cases with them.